testnet/stacks-node/src/tests/integrations.rs

```diff
@@ -713,6 +713,53 @@ fn contract_stx_transfer() {
     run_loop.start(num_rounds);
 }
 
+/// Test mining a smart contract twice (in non-sequential blocks)
```

Comment:
Can this bug happen if someone tries publish the same contract twice in the same block, or is that caught already by some other means?  Also, why did this bug not manifest in sequential blocks?

Reply:
No, this bug can't happen if someone tries to publish the same contract twice in the same block -- Clarity returns a check error if you try to publish the same contract twice (and that's tested in a couple of other unit and integration tests). This is also why this bug won't be manifest in sequential blocks -- the state database has the information required to reject the transaction before it processes it.

This error only occurs when we attempt to move the sidecar metadata on miner's block -- the miner is correctly able to process the transaction, since the contract doesn't exist in the chainstate it's building off of, but when it tries to move the metadata's block hash to '{miner-block-hash}.mined', it finds that there's already data there (since every mined block uses the same identifier).

Reply:
Do you mean "side storage metadata"? ;)

This explanation makes sense to me.  Thanks!